src/lib/mdns/ServiceNaming.cpp

```diff
@@ -115,7 +115,7 @@ CHIP_ERROR MakeHostName(char * buffer, size_t bufferLen, const chip::ByteSpan &
     size_t idx = 0;
     for (size_t i = 0; i < macOrEui64.size(); ++i)
     {
-        idx += snprintf(buffer + idx, 3, "%02X", macOrEui64.data()[i]);
+        idx += snprintf(buffer + idx, 3, "%X", macOrEui64.data()[i]);
```

Comment:
I don't think this is correct. I think hostname is still fixed-length per current spec.

Reply:
Hostname is fixed length. We also could use BytesToHex.h for this particular conversion here, it is less flash overall than this large callsite of a variadic function.

Reply:
In the interest of time, it's probably easiest to just put %02X back in here and do the BytesToHex later

Reply:
fixed in next commit